packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -443,10 +442,47 @@ class _HeroFlightManifest {
     );
   }
 
+  Tween<Rect?> createHeroRectTween({ required Rect? begin, required Rect? end }) {
+    final CreateRectTween? createRectTween = toHero.widget.createRectTween ?? this.createRectTween;
+    return createRectTween?.call(begin, end) ?? RectTween(begin: begin, end: end);
+  }
+
+  // The bounding box for `context`'s render object,  in `ancestorContext`'s
+  // render object's coordinate space.
+  static Rect _boundingBoxFor(BuildContext context, BuildContext? ancestorContext) {
+    assert(ancestorContext != null);
+    final RenderBox box = context.findRenderObject()! as RenderBox;
+    assert(box != null && box.hasSize && box.size.isFinite);
```

Comment:
if the fromHeroLocation.size is not finite, this assertion will throw before isValid returns false?

Reply:
The PR for the most part is trying to address infinity/NaN introduced by applying a zero matrix. If RenderBox.size contains either infinity/NaN then it's probably an error.

Reply:
this maybe an extreme case, I have encountered a use case where there are renderobject with zero size. One customer introduce a widget that has various size depends on data load. not that I think we should definitely handle this case, but would like to bring that up for your attention.

Reply:
Sorry I fail to understand the concern here, are you saying if the bounding box is zero-sized it's going to cause problems?

Reply:
If the renderobject is zero size I think this will throw before the hero flight manifest isvalid returns false

Reply:
How so? `Size.zero` seems valid to me, isn't `Size.zero.isFinite` true?

Reply:
for some reason i keep thinking isfinite = size >0 <infinite, yeah i think this is good